public/app/plugins/panel/timeseries/plugins/OutsideRangePlugin.tsx

```diff
@@ -34,11 +34,26 @@ export const OutsideRangePlugin: React.FC<ThresholdControlsPluginProps> = ({ con
   }
 
   // Time values are always sorted for uPlot to work
-  const first = timevalues[0];
-  const last = timevalues[timevalues.length - 1];
+  let i = 0,
+    j = timevalues.length - 1;
+
+  while (i < timevalues.length && timevalues[i] == null) {
```

Comment:
can likely be shortened ;)

```suggestion
  while (i <= j && timevalues[i] == null) {
```